basis/cpu/x86/64/unix/unix.factor

```diff
@@ -24,16 +24,27 @@ M: x86.64 reserved-stack-space 0 ;
         [ 8 mod zero? [ t , ] when , ] assoc-each
     ] { } make { t } split harvest ;
 
-: flatten-small-struct ( c-type -- seq )
-    struct-types&offset split-struct [
+:: flatten-small-struct ( c-type -- seq )
+    c-type struct-types&offset split-struct [
         [ lookup-c-type c-type-rep reg-class-of ] map
         int-regs swap member? int-rep double-rep ?
         f f 3array
-    ] map ;
+    ] map :> reps
+    int-reg-reps get float-reg-reps get and [
+        reps [ first int-rep? ] count :> int-mems
+        reps length int-mems - :> float-mems
+        int-reg-reps get int-mems + 6 >
+        float-reg-reps get float-mems + 8 > or [
+            reps [ first t f 3array ] map
+        ] [ reps ] if
+    ] [ reps ] if ;
 
 M: x86.64 flatten-struct-type ( c-type -- seq )
     dup heap-size 16 <=
-    [ flatten-small-struct ] [ call-next-method [ first t f 3array ] map ] if ;
+    [ flatten-small-struct record-reg-reps ] [
+        call-next-method [ first t f 3array ] map
+        unrecord-reg-reps
```

Comment:
Isn't this doing no work since ``second`` is always ``t``?  (Due to ``[ first t f 3array ] map``)? 

So ``unrecord-reg-reps`` will not decrement anything?

Reply:
Indeed, this is obviously strange. I ponder what I wanted to do. 
Please wait for a while to consider other points too.

Reply:
If the structure is larger than 16 bytes, Factor doesn't need to `record-reg-reps`, so change the above code.
```Factor
M: x86.64 flatten-struct-type ( c-type -- seq )
    dup heap-size 16 <=
    [ flatten-small-struct record-reg-reps ] [
        call-next-method unrecord-reg-reps
        [ first t f 3array ] map
    ] if ;
```